ament_pep257/ament_pep257/main.py

```diff
@@ -187,13 +187,24 @@ def generate_pep257_report(paths, excludes, ignore, select, convention, add_igno
     report = []
 
     files_dict = {}
-    for filename, checked_codes, ignore_decorators in files_to_check:
-        if _filename_in_excludes(filename, excludes):
-            continue
-        files_dict[filename] = {
-            'select': checked_codes,
-            'ignore_decorators': ignore_decorators,
-        }
+    # Unpack 3 values for pydocstyle <= 6.1.1
+    try:
+        for filename, checked_codes, ignore_decorators in files_to_check:
+            if _filename_in_excludes(filename, excludes):
+                continue
+            files_dict[filename] = {
+                'select': checked_codes,
+                'ignore_decorators': ignore_decorators,
+            }
+    # Unpack 4 values since pydocstyle >= 6.2.0
+    except ValueError:
+        for filename, checked_codes, ignore_decorators, _ in files_to_check:
+            if _filename_in_excludes(filename, excludes):
+                continue
+            files_dict[filename] = {
+                'select': checked_codes,
+                'ignore_decorators': ignore_decorators,
+            }
```

Comment:
This definitely works, but duplicates the body of the loop.

I actually just learned about a new Python 3 syntax that lets you unpack an optional set of arguments at the end, `*_`.  So I think we can handle returning 3 or 4 tuple items, without duplicating the body of the loop, with:

```suggestion
    # Unpack 3 values for pydocstyle <= 6.1.1, and 4 values for pydocstyle >= 6.2.0.
    for filename, checked_codes, ignore_decorators, *_ in files_to_check:
        if _filename_in_excludes(filename, excludes):
            continue
        files_dict[filename] = {
            'select': checked_codes,
            'ignore_decorators': ignore_decorators,
        }
```

Can you give this a try and see if it still solves the problem?